gradle-plugin/plugin/src/main/java/com/yelp/codegen/plugin/GenerateTask.kt

```diff
@@ -20,59 +26,65 @@ const val DEFAULT_NAME = "defaultname"
 const val DEFAULT_PACKAGE = "com.codegen.default"
 const val DEFAULT_OUTPUT_DIR = "/gen"
 
-open class GenerateTask : DefaultTask() {
+abstract class GenerateTask : DefaultTask() {
```

Comment:
Making this class abstract allows to leverage Gradle service injection to create Properties automagically: https://docs.gradle.org/current/userguide/custom_gradle_types.html#service_injection